oomagent/server.go

```diff
@@ -292,18 +295,55 @@ func (s *server) ChannelJoin(stream codegen.OomAgent_ChannelJoinServer) error {
 }
 
 func (s *server) Join(ctx context.Context, req *codegen.JoinRequest) (*codegen.JoinResponse, error) {
-	err := s.oomstore.Join(ctx, types.JoinOpt{
-		FeatureNames:   req.Features,
-		InputFilePath:  req.InputFile,
-		OutputFilePath: req.OutputFile,
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	joinResult, err := s.oomstore.Join(ctx, types.JoinOpt{
+		FeatureNames:  req.Features,
+		InputFilePath: req.InputFile,
 	})
 	if err != nil {
 		return nil, internalError(err.Error())
 	}
 
+	if err := writeJoinResultToFile(req.OutputFile, joinResult); err != nil {
+		return nil, wrapErr(err)
+	}
+
 	return &codegen.JoinResponse{}, nil
 }
 
+func writeJoinResultToFile(outputFilePath string, joinResult *types.JoinResult) error {
```

Comment:
can we reuse `writeJoinResultToFile`, we already have the same one in `oomstore` package

Reply:
`writeJoinResultToFile` Currently only oomagent is in use, so I migrated `writeJoinResultToFile` to   
oomagent layer.

Reply:
gotcha!